### ISSUE TITLE
Chapter 4: n_inputs in place of n_outputs

### ISSUE THREAD

**Issue by carlocav** [ageron/handson-ml#125]
In the _gradients =_ lines in [76] and [78] (Exercise solutions 12. Batch Gradient Descent with early stopping for Softmax Regression), I think n_inputs should be replaced with n_outputs. Anyway the output doesn't change because in this situation n_inputs is equal to n_outputs.

**Comment by ageron** [comment]
Good catch @carlocav ! Since `n_inputs` is equal to  `n_outputs` in this case, it did not cause an error, so I missed it.  I just fixed it.
Thanks a lot!  :)
